src/main/kotlin/org/rust/cargo/project/workspace/CargoWorkspace.kt

```diff
@@ -211,14 +209,30 @@ private class WorkspaceImpl(
         // so we have to rebuild the whole workspace from scratch instead of
         // *just* adding in the stdlib.
 
-        val stdAll = stdlib.crates.associateBy { it.id }
-        val stdGated = stdlib.crates.filter { it.type == StdLibType.FEATURE_GATED }.map { it.id }.toSet()
-        val stdRoots = stdlib.crates.filter { it.type == StdLibType.ROOT }.map { it.id }.toSet()
+        val (newPackagesData, stdCrates) = if (!stdlib.isPartOfCargoProject) {
```

Comment:
These changes are about better `rustc` project support: now we don't duplicate stdlib packages (because they are already a part of the project)